live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/filter/AspectFilterApplier.java

```diff
@@ -59,7 +59,7 @@ public boolean applyFilter(RepoEvent<DataAttributes<NodeResource>> repoEvent, Fi
 
     private boolean isAllowed(Set<String> aspectNames, List<String> allowed)
     {
-        return CollectionUtils.isEmpty(allowed) || allowed.stream().anyMatch(aspectNames::contains);
+        return allowed.isEmpty() || !aspectNames.isEmpty() && new HashSet<>(allowed).containsAll(aspectNames);
```

Comment:
I don't think this is correct:
```
aspectNames: a, b
allowed: a, c
allowed.containsAll(aspectNames) -> false
```
In this case I'd expect the presences of aspect `a` to qualify the node.

Reply:
Reverted